### PR TITLE
generate valid IDs for candidates in CDF JSON

### PR DESCRIPTION
**Pull request by tarheel**
As requested, we now generate a valid JSON ID for each candidate and include the full candidate name as a separate value.

### CODE REVIEW

File: src/main/java/com/rcv/ResultsWriter.java [hunk 1]
```diff
@@ -56,6 +56,8 @@
   private static final String CDF_GPU_ID = "gpu-001";
   private static final String CDF_REPORTING_DEVICE_ID = "rd-001";
 
+  private static final Map<String, String> candidateCodeToJsonId = new HashMap<>();
```
Comment (moldover) [inline src/main/java/com/rcv/ResultsWriter.java]: Would it be clearer to change JsonId => CdfId ?   Here and elsewhere...

File: src/main/java/com/rcv/ResultsWriter.java [hunk 2]
```diff
@@ -607,23 +622,29 @@ void generateCdfJson(List<CastVoteRecord> castVoteRecords) throws IOException {
...
-    List<Map<String, String>> contestSelections = new LinkedList<>();
-    for (String candidate : config.getCandidateCodeList()) {
-      contestSelections.add(
-          Map.ofEntries(entry("@id", candidate), entry("@type", "CVR.ContestSelection")));
+    List<Map<String, Object>> contestSelections = new LinkedList<>();
+    for (String candidateCode : config.getCandidateCodeList()) {
+      Map<String, String> codeMap = Map.ofEntries(
+          entry("@type", "CVR.Code"),
+          entry("Type", "other"),
+          entry("OtherType", "vendor-label"),
```
Comment (moldover) [inline src/main/java/com/rcv/ResultsWriter.java]: vendor-label doesn't seem right

File: src/main/java/com/rcv/ResultsWriter.java (hunk 1 above)
Comment (tarheel) [reply src/main/java/com/rcv/ResultsWriter.java]: Sure.

File: src/main/java/com/rcv/ResultsWriter.java (hunk 2 above)
Comment (tarheel) [reply src/main/java/com/rcv/ResultsWriter.java]: I'm just following what John suggested in his email.

Comment (moldover) [reply src/main/java/com/rcv/ResultsWriter.java]: In that case, great :)

**Pull request merged**